File: src/main/java/com/demo/t_web/program/ai/component/ChatClientComponent.java
package com.demo.t_web.program.ai.component;

import lombok.Getter;
import org.springframework.ai.chat.client.ChatClient;
import org.springframework.stereotype.Component;

/**
 * <pre>
 * com.demo.t_web.program.ai.component.ChatClientService
 *   - ChatClientService.java
 * </pre>
 *
 * @author : tarr4h
 * @className : ChatClientService
 * @description :
 * @date : 12/28/24
 */
@Component
@Getter
public class ChatClientComponent {

    private final ChatClient chatClient;

    public ChatClientComponent(ChatClient.Builder builder){
        this.chatClient = builder.build();
    }
}

File: src/main/java/com/demo/t_web/program/ai/component/ChatClientComponent.kt
package com.demo.t_web.program.ai.component

import org.springframework.ai.chat.client.ChatClient
import org.springframework.ai.chat.client.advisor.MessageChatMemoryAdvisor
import org.springframework.ai.chat.memory.InMemoryChatMemory
import org.springframework.stereotype.Component

@Component
class ChatClientComponent(
    builder : ChatClient.Builder
) {
    val chatClient : ChatClient = builder
        .defaultAdvisors(MessageChatMemoryAdvisor(InMemoryChatMemory()))
        .build()
}